From jeffm@suse.com  Mon Aug 10 16:19:53 2009
From: Jeff Mahoney <jeffm@suse.com>
Date: Tue, 04 Aug 2009 10:26:19 -0400
Subject: Staging: rtl8192e: compile fixes
To: Greg KH <gregkh@suse.com>
Message-ID: <20090804142709.911355735@suse.com>


This patch removes -fhard-float and the software float helpers. In-kernel
floating point is not allowed.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192e/Makefile      |    1 -
 drivers/staging/rtl8192e/r8192E_core.c |    8 --------
 2 files changed, 9 deletions(-)

--- a/drivers/staging/rtl8192e/Makefile
+++ b/drivers/staging/rtl8192e/Makefile
@@ -4,7 +4,6 @@ NIC_SELECT = RTL8192E
 EXTRA_CFLAGS += -DRTL8192E
 EXTRA_CFLAGS += -std=gnu89
 EXTRA_CFLAGS += -O2
-EXTRA_CFLAGS += -mhard-float -DCONFIG_FORCE_HARD_FLOAT=y
 EXTRA_CFLAGS += -DTHOMAS_TURBO
 EXTRA_CFLAGS += -DENABLE_DOT11D
 
--- a/drivers/staging/rtl8192e/r8192E_core.c
+++ b/drivers/staging/rtl8192e/r8192E_core.c
@@ -24,14 +24,6 @@
  * Jerry chuang <wlanfae@realtek.com>
  */
 
-#ifndef CONFIG_FORCE_HARD_FLOAT
-double __floatsidf (int i) { return i; }
-unsigned int __fixunsdfsi (double d) { return d; }
-double __adddf3(double a, double b) { return a+b; }
-double __addsf3(float a, float b) { return a+b; }
-double __subdf3(double a, double b) { return a-b; }
-double __extendsfdf2(float a) {return a;}
-#endif
 
 #undef LOOP_TEST
 #undef RX_DONT_PASS_UL
